pkgs/desktops/cinnamon/cinnamon-desktop/default.nix

```diff
@@ -0,0 +1,68 @@
+{ fetchFromGitHub
+, gdk-pixbuf
+, gobject-introspection
+, gtk3
+, intltool
+, meson
+, ninja
+, pkgconfig
+, pulseaudio
+, python3
+, stdenv
+, xkeyboard_config
+, xorg
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cinnamon-desktop";
+  version = "4.4.0";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "17hb8gkb9pfj56ckva5g4x83yvmdv7hvpidxjsdf79dw6pabr5rg";
+  };
+
+  buildInputs = [
+    gdk-pixbuf
+    gtk3
+    pulseaudio
+    xkeyboard_config
+    xorg.libxkbfile
+    gobject-introspection
+  ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    python3
+    wrapGAppsHook
+    intltool
+    pkgconfig
+  ];
+
+  postPatch = ''
+    chmod +x install-scripts/meson_install_schemas.py # patchShebangs requires executable file
+    patchShebangs install-scripts/meson_install_schemas.py
+    sed "s|/usr/share|/run/current-system/sw/share|g" -i ./schemas/*
```

Comment:
We should try to hardcode these paths into the schemas from which package they come from.
I'm seeing
* https://github.com/linuxmint/cinnamon-desktop/blob/b823080f53b3b32afb0e0956ba2a23e2644c06e3/schemas/org.cinnamon.desktop.a11y.keyboard.gschema.xml.in.in#L114
* https://github.com/linuxmint/cinnamon-desktop/blob/b823080f53b3b32afb0e0956ba2a23e2644c06e3/schemas/org.cinnamon.desktop.wm.preferences.gschema.xml.in.in#L302

I'm not sure what package these sounds come from, so I guess you can leave them unpatched and fix them when you add the package. (guessing cinnamon-sounds?)

Reply:
It's from cinnamon-common which is the root package

Circular dependency if we patch this properly?

```
maciej@mkg-lenovo:~/Projekte/nixpkgs$ readlink -f /usr/share/cinnamon/sounds/
/nix/store/k6plfqprqx4600cz7v623knflqkzqzdi-cinnamon-common-4.4.1/share/cinnamon/sounds
```

(also don't ask me about the dark magic that made /usr/share a thing xD)

Reply:
> It's from cinnamon-common which is the root package

Do you mean it's at the top of the dependency tree?
If so then it shouldn't be an issue, but if `ciinnamon-common` depends on `cinnamon-desktop` it will be circular, and we'd have to ask cinnamon to ship those files somewhere else.